org.lflang/src/org/lflang/ASTUtils.java

```diff
@@ -1556,60 +1578,68 @@ static def withoutQuotes(String s) {
-        val main = resource.allContents.filter(Reactor).findFirst[it.isMain || it.isFederated]
-        if (main !== null && main.name.isNullOrEmpty) {
-            main.name = name
+    public static void setMainName(Resource resource, String name) {
+        Reactor main = IteratorExtensions.findFirst(Iterators.filter(resource.getAllContents(), Reactor.class), 
+            it -> { return it.isMain() || it.isFederated(); }
+        );
+        if (main != null && StringExtensions.isNullOrEmpty(main.getName())) {
+            main.setName(name);
         }
     }
     
     /**
      * Create a new instantiation node with the given reactor as its defining class.
      * @param reactor The reactor class to create an instantiation of.
      */
-    def static createInstantiation(Reactor reactor) {
-        val inst = LfFactory.eINSTANCE.createInstantiation
-        inst.reactorClass = reactor
+    public static Instantiation createInstantiation(Reactor reactor) {
+        Instantiation inst = LfFactory.eINSTANCE.createInstantiation();
+        inst.setReactorClass(reactor);
         // If the reactor is federated or at the top level, then it
         // may not have a name. In the generator's doGenerate()
         // method, the name gets set using setMainName().
         // But this may be called before that, e.g. during
         // diagram synthesis.  We assign a temporary name here.
-        if (reactor.name === null) {
-            if (reactor.isFederated || reactor.isMain) {
-                inst.setName("main")    
+        if (reactor.getName() == null) {
+            if (reactor.isFederated() || reactor.isMain()) {
+                inst.setName("main");
             } else {
-                inst.setName("")
+                inst.setName("");
             }
             
         } else {
-            inst.setName(reactor.name)
+            inst.setName(reactor.getName());
         }
-        return inst
+        return inst;
     }
 
     /**
      * Returns the target declaration in the given model.
      * Non-null because it would cause a parse error.
      */
-    def static TargetDecl targetDecl(Model model) {
-        return model.eAllContents.filter(TargetDecl).head
+    public static TargetDecl targetDecl(Model model) {
+        return IteratorExtensions.head(Iterators.filter(model.eAllContents(), TargetDecl.class));
     }
 
     /**
      * Returns the target declaration in the given resource.
      * Non-null because it would cause a parse error.
      */
-    def static TargetDecl targetDecl(Resource model) {
-        return model.allContents.filter(TargetDecl).head
+    public static TargetDecl targetDecl(Resource model) {
+        return IteratorExtensions.head(Iterators.filter(model.getAllContents(), TargetDecl.class));
     }
 
+    /**
+     * Returns the list if it is not null. Otherwise return an empty list.
+     */
+    private static <T> List<T> convertToEmptyListIfNull(List<T> list) {
```

Comment:
I'm OK with leaving this here for now, but it should probably go into a separate `ListUtil` class. There's also `ListUtils.emptyIfNull(List<T> list)` in Apache Collections 4. I'm not sure if we already have that as a dependency in the project, but if we do, we should just use that.

Reply:
The project currently does not have apache collections4 as a dependency